iroh-api/src/p2p.rs

```diff
@@ -56,6 +65,25 @@ impl P2p {
             .await
             .map_err(|e| map_service_error("p2p", e))
     }
+
+    pub async fn network_events(&self) -> Result<BoxStream<'static, Result<NetworkEvent>>> {
+        let stream = self
+            .client
+            .network_events()
+            .await
+            .map_err(|e| map_service_error("p2p", e))?;
+        Ok(stream.boxed())
+    }
+
+    pub async fn subscribe(&self, topic: String) -> Result<bool> {
+        let topic = TopicHash::from_raw(topic);
+        self.client.gossipsub_subscribe(topic).await
+    }
+
+    pub async fn publish(&self, topic: String, data: Bytes) -> Result<MessageId> {
+        let topic = TopicHash::from_raw(topic);
+        self.client.gossipsub_publish(topic, data).await
+    }
```

Comment:
Thought I'd leave these as `subscribe`, and `publish` rather than `gossipsub_subscribe` and `gossipsub_publish`, to keep the API a little slicker looking. But is it too confusing/vague?

Reply:
In any case I think every function here should have a looong doc comment describing what it does and providing some examples.  :)

I'd probably be tempted to leave the `gossipsub` part it, maybe there will be other things to subscribe to in the future?  If leaving it out there's only one canonical way to subscribe a p2p node to topics.  But that's speaking as someone who has no experience using these things.  I don't really know the domain well enough to make a call on this kind of API naming.